src/vs/workbench/services/dialogs/electron-browser/dialogService.ts

```diff
@@ -224,7 +228,7 @@ export class FileDialogService implements IFileDialogService {
 			options.defaultUri = this.defaultFilePath(schema);
 		}
 
-		if (schema !== Schemas.file) {
+		if ((schema !== Schemas.file) || (this.configurationService.getValue(useSimplifiedDialogKey) === 'true')) {
 			const title = nls.localize('openFileOrFolder.title', 'Open File Or Folder');
 			const availableFileSystems = [schema, Schemas.file]; // always allow file as well
```

Comment:
change to something like `schema !== Schema.file ? [ schema, Schema.file] : [ schema]`